### PR TITLE
Add dm_thin_pool for gluster use

### LINKED ISSUE #5108: Glusterfs: unable install if docker storage driver is overlay2
I know how to fix it manually, but I don't want to use device-mapper, I think no one want :)

OS: 
`Fedora Atomic 26` 

ansible:
```sh
ansible --version
ansible 2.3.1.0
  config file = /usr/share/ansible/openshift-ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.5 (default, Nov  6 2016, 00:28:07) [GCC 4.8.5 20150623 (Red Hat 4.8.5-11)]
```
inventory:
```sh
[OSEv3:children]
masters
nodes
etcd
glusterfs

[OSEv3:vars]
ansible_ssh_user=fedora
ansible_become=true
ansible_ssh_transfer_method=scp
deployment_type=origin
containerized=true
openshift_master_cluster_method=native
openshift_master_cluster_password=admin
openshift_release=3.6
openshift_image_tag=v3.6.0
openshift_router_selector='region=infra'
openshift_registry_selector='region=infra'

openshift_install_examples=false

glusterfs_wipe=True
openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
openshift_schedulable=true
#monitoring

# Enable cockpit
osm_use_cockpit=true

# Set cockpit plugins
osm_cockpit_plugins=['cockpit-kubernetes']

openshift_hosted_logging_deploy=true
openshift_hosted_metrics_deploy=true
openshift_set_hostname=True

openshift_disable_check=disk_availability, docker_storage, memory_availability

openshift_master_default_subdomain=test2.com
openshift_master_cluster_hostname=test2.com
openshift_master_cluster_public_hostname=test2.com

[etcd]
sir-98p8j2dp
sir-2pq8g33p
sir-6z8gkj5n

[masters]
sir-98p8j2dp
sir-2pq8g33p
sir-6z8gkj5n

[nodes]
sir-98p8j2dp
sir-2pq8g33p
sir-6z8gkj5n

[glusterfs]
sir-98p8j2dp glusterfs_devices='[ "/dev/xvdb" ]'
sir-2pq8g33p glusterfs_devices='[ "/dev/xvdb" ]'
sir-6z8gkj5n glusterfs_devices='[ "/dev/xvdb" ]'
``` 

error what I got:
```sh
TASK [openshift_storage_glusterfs : Load heketi topology] **********************
changed: [sir-w18rjtnn] => {
    "changed": true, 
    "cmd": [
        "/usr/local/bin/oc", 
        "rsh", 
        "--namespace=glusterfs", 
        "deploy-heketi-storage-1-xtfpf", 
        "heketi-cli", 
        "-s", 
        "http://localhost:8080", 
        "--user", 
        "admin", 
        "--secret", 
        "mJKFHcKtImYUhezlenouYwAf8Q48lFgABaTxoX4paJc=", 
        "topology", 
        "load", 
        "--json=/tmp/openshift-glusterfs-ansible-qg81xw/topology.json", 
        "2>&1"
    ], 
    "delta": "0:00:08.833600", 
    "end": "2017-08-16 09:18:33.270592", 
    "failed": false, 
    "failed_when_result": false, 
    "rc": 0, 
    "start": "2017-08-16 09:18:24.436992"
}

STDOUT:

Creating cluster ... ID: 0a294f18b26891e9e83793146bb6e195
  Creating node ip-10-0-1-145.eu-central-1.compute.internal ... ID: 12f2ff53810234e04cdc034f69d39335
    Adding device /dev/xvdb ... OK
  Creating node ip-10-0-1-82.eu-central-1.compute.internal ... ID: 44fdb837f77a49341ce7878fd7391ac9
    Adding device /dev/xvdb ... OK
  Creating node ip-10-0-1-26.eu-central-1.compute.internal ... ID: 8dfe002d322470c3286be6fdbd5d8aee
    Adding device /dev/xvdb ... OK

TASK [openshift_storage_glusterfs : Create heketi DB volume] *******************
fatal: [sir-w18rjtnn]: FAILED! => {
    "changed": true, 
    "cmd": [
        "/usr/local/bin/oc", 
        "rsh", 
        "--namespace=glusterfs", 
        "deploy-heketi-storage-1-xtfpf", 
        "heketi-cli", 
        "-s", 
        "http://localhost:8080", 
        "--user", 
        "admin", 
        "--secret", 
        "mJKFHcKtImYUhezlenouYwAf8Q48lFgABaTxoX4paJc=", 
        "setup-openshift-heketi-storage", 
        "--listfile", 
        "/tmp/heketi-storage.json"
    ], 
    "delta": "0:00:10.515663", 
    "end": "2017-08-16 09:18:44.627313", 
    "failed": true, 
    "rc": 255, 
    "start": "2017-08-16 09:18:34.111650"
}

STDERR:

Error: Unable to execute command on glusterfs-storage-ljvpr:   /usr/sbin/modprobe failed: 1
  thin: Required device-mapper target(s) not detected in your kernel.
  Run `lvcreate --help' for more information.
command terminated with exit code 255
```

### PR DESCRIPTION
**Pull request by sdodson**
Fixes #5108 
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1490905

### CODE REVIEW

File: roles/openshift_node/templates/openshift-cns.conf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#{{ ansible_managed }}
+dm_thin_pool
```
Comment (jarrpa) [inline roles/openshift_node/templates/openshift-cns.conf]: You need to also add `dm_snapshot` and `dm_mirror`.

File: roles/openshift_node/tasks/storage_plugins/glusterfs.yml [hunk 2]
```diff
@@ -52,3 +52,9 @@
   - ('--> off' in item.stdout)  # Boolean is currently off.
   - ansible_python_version | version_compare('3', '>=')
   with_items: "{{ fusefs_getsebool_status.results }}"
+  
+- name: Ensure device mapper modules loaded
+  template:
+    src: openshift-cns.conf
+    path: /etc/modules-load.d/openshift-cns.conf
```
Comment (jarrpa) [inline roles/openshift_node/tasks/storage_plugins/glusterfs.yml]: Rename to something like `glusterfs.conf` or `kmod-glusterfs.conf`. :)

**Comment by jarrpa** [comment]
@sdodson Wait wait wait, I missed something: these kernel modules are not needed for the storage plugin, they're needed for the GlusterFS service. You don't need to enable these modules on every OCP node, just the ones that will be running GlusterFS pods. Hence, I think these changes belong in the GlusterFS role and not in the node role.

**Comment by sdodson** [comment]
@jarrpa ok, updated for both comments

**Comment by jarrpa** [comment]
@sdodson My third comment about the GlusterFS role is sitll not addressed.

**Comment by jarrpa** [comment]
OH, sorry, didn't understand what was triggered with the tag I used. :(

General LGTM, then!

**Comment by jarrpa** [comment]
@sdodson Agh, forgot to approve the changes. Is this ready to go in now, or does it still need human testing?

File: roles/openshift_storage_glusterfs/tasks/kernel_modules.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+- name: Ensure device mapper modules loaded
+  template:
+    src: glusterfs.conf
+    path: /etc/modules-load.d/glusterfs.conf
```
Comment (dustymabe) [inline roles/openshift_storage_glusterfs/tasks/kernel_modules.yml]: should be `dest`, not `path`.

**Comment by dustymabe** [comment]
ok after I changed `path` to `dest` everything works like expected. update this PR with that change and LGTM.

**Pull request merged**